mipengine/node_registry/node_registry.py

```diff
@@ -0,0 +1,181 @@
+import consul
```

Comment:
The Node Registry is a separate service/process so I want its files to be separate from the other services' files of the system. I would prefer to leave it like that for now and reexamine it when we have a proposal for the different folder structure of the files of the whole system
Concerning the DTOs exchanged between one service(node or controller) and another service, namely the node registry, I do not think we should put all the DTOs we use in the system in one big ..DTOs... file. I believe they should be in separate files based on the service they are used for. But again I believe this is a separate conversation and decision to make.

Reply:
I think there won't be a "big proposal" for a restructure. I remember we said that each of us will try to reduce the nesting in his parts of the code.

I agree that the "node registry" is a service but the `node_registry.py` is an adapter/interface/client not the code of the service. Since right now at least is just one file it could be in the mipengine folder directly `mipengine/node_registry.py`, why nest it inside a folder? If you like it that way, I am fine with it off course.

About the DTOs. My point wasn't to add them in one big DTOs file. My point was to put them inside the `node_registry.py` file. In my opinion there is no point of having them in a separate file since whenever you will want to use them you will import the `node_registry.py` as well anyway. Am I missing something?

The DTOs in other parts of the system are separate from the service because we don't want to import the whole service to send a request. But the `node_registry` is a client not a service so it makes sense to have the DTOs in that module, they are parts of the same thing.

Reply:
Yes, you re right the dtos related to the node registry service do not need to be in a different file, I ll join node_registry_DTOs.py and node_registry.py and just throw it in the mipengine/ folder